src/test/java/org/folio/fqm/service/PermissionsRegularServiceTest.java

```diff
@@ -134,9 +137,31 @@ void shouldCheckFqmPermissionsIfRequested() {
   }
 
   @Test
-  void eurekaSupportIsNotImplementedYet() {
-    permissionsService.isEureka = true; // Force the service to use mod-roles-keycloak
-    setUpMocks();
-    assertThrows(NotImplementedException.class, () -> permissionsService.getUserPermissions(TENANT_ID));
+  void WhenEurekaIsTrue() {
+    permissionsService.isEureka = true;
+    setUpMocks("permission1", "permission2");
+    Set<String> userPermissions = permissionsService.getUserPermissions();
+
+    ArgumentCaptor<UUID> userIdCaptor = ArgumentCaptor.forClass(UUID.class);
+    verify(modRolesKeyclockClient).getPermissionsUser(eq(TENANT_ID), userIdCaptor.capture());
+    assertEquals(2, userPermissions.size());
+
+    // Ensure no interactions with modPermissionsClient
+    verifyNoInteractions(modPermissionsClient);
+  }
+
+  @Test
+  void WhenEurekaIsFalse() {
+    permissionsService.isEureka = false;
+    setUpMocks("permission1", "permission2");
+    Set<String> userPermissions = permissionsService.getUserPermissions();
+
+    ArgumentCaptor<String> userIdCaptor = ArgumentCaptor.forClass(String.class);
+    verify(modPermissionsClient).getPermissionsForUser(eq(TENANT_ID), userIdCaptor.capture());
+    assertEquals(2, userPermissions.size());
+
+    // Ensure no interactions with modRolesKeyclockClient;
```

Comment:
stupid nit: please remove the `;` to make sonar happy (it thinks this is a line of code)
```suggestion
    // Ensure no interactions with modRolesKeyclockClient
```